Documentation/ApiOverview/Authentication/CSRFlikeRequestTokenHandling.rst

```diff
@@ -0,0 +1,163 @@
+    supports five different nonces in the same request. The pool purges nonces
+    15 minutes (900 seconds) after they have been issued.
+
+Workflow
+========
+
+The sequence looks like the following:
+
+..  rst-class:: bignums-xxl
+
+#.  Retrieve nonce and request token values
+
+    This happens on the previous legitimate visit on a page that offers
+    a corresponding form that shall be protected. The `RequestToken` and `Nonce`
+    objects (later created implicitly in this example) are organized in the
+    :php:`\TYPO3\CMS\Core\Context\SecurityAspect`.
+
+    ..  code-block:: php
+        :caption: EXT:my_extension/Classes/Controller/MyController.php
+
+        use TYPO3\CMS\Core\Security\RequestToken;
+        use TYPO3\CMS\Fluid\View\StandaloneView;
+
+        final class MyController
+        {
+            private StandaloneView $view;
+
+            public function showFormAction()
+            {
+                // creating new request token with scope 'my/process' and hand over to view
+                $requestToken = RequestToken::create('my/process');
+                $this->view->assign('requestToken', $requestToken);
+                // ...
+            }
+
+            public function processAction() {
+                // for the implementation, see below
+            }
+        }
+
+    ..  code-block:: html
+        :caption: EXT:my_extension/Resources/Private/Templates/ShowForm.html
+
+        <!-- Assign request token object for ViewHelper -->
+        <f:form action="process" requestToken="{requestToken}>
+            ...
+        </f:form>
+
+    The HTTP response on calling the shown controller action above will be like
+    this:
+
+    ..  code-block:: text
+
+        HTTP/1.1 200 OK
+        Content-Type: text/html; charset=utf-8
+        Set-Cookie: typo3nonce_[hash]=[nonce-as-jwt]; path=/; httponly; samesite=strict
+
+        ...
+        <form action="/my/process" method="post">
+            ...
+            <input type="hidden" name="__request_token" value="[request-token-as-jwt]">
+            ...
+        </form>
+
+#.  Invoke action request and provide nonce and request token values
+
+    When submitting the form and invoking the corresponding action, same-site
+    cookies `typo3nonce_[hash]` and request-token value `__RequestToken` are
+    sent back to the server. Without using a separate nonce in a scope that is
+    protected by the client, the corresponding request token could be easily
+    extracted from markup and used without having the possibility to verify the
+    procedural integrity.
+
+    The middleware :php:`\TYPO3\CMS\Core\Middleware\RequestTokenMiddleware`
+    takes care of providing the received nonce and received request token values
+    in :php:`\TYPO3\CMS\Core\Context\SecurityAspect`. The handling controller
+    action needs to verify that the request token has the expected
+    `'my/process'` scope.
+
+    ..  code-block:: php
```

Comment:
move this to a file?

Reply:
Thought about it, but it is not really valid (the view is not initialized here). Can do that, of course, but then we have more distracting code from the substantial things.